tests/unit/amo/components/TestSearchContextCard.js

```diff
@@ -111,162 +134,393 @@ describe(__filename, () => {
+        addonType: ADDON_TYPE_DICT,
+      },
+    });
+
+    const root = render({ store });
+
+    expect(root.find('.SearchContextCard-header')).toIncludeText(
+      '2 results found',
     );
   });
 
   it('should render singular form when only one result is found with addonType ADDON_TYPE_EXTENSION', () => {
+    const query = 'test';
     const { store } = dispatchSearchResults({
       addons: { [fakeAddon.slug]: fakeAddon },
       filters: {
         addonType: ADDON_TYPE_EXTENSION,
-        query: 'test',
+        query,
       },
     });
 
     const root = render({ store });
 
     expect(root.find('.SearchContextCard-header')).toIncludeText(
-      '1 extension found for "test"',
+      `1 extension found for "${query}"`,
     );
   });
 
   it('should render plural form when multiple results are found with addonType ADDON_TYPE_EXTENSION', () => {
+    const query = 'test';
     const { store } = dispatchSearchResults({
       filters: {
         addonType: ADDON_TYPE_EXTENSION,
-        query: 'test',
+        query,
+      },
+    });
+
+    const root = render({ store });
+
+    expect(root.find('.SearchContextCard-header')).toIncludeText(
+      `2 extensions found for "${query}"`,
+    );
+  });
+
+  it('should render results with categoryName and query for addonType ADDON_TYPE_EXTENSION when loading is false', () => {
+    const query = 'test';
+    const categoryName = 'Causes';
+    const { store } = dispatchSearchResults({
+      filters: {
+        addonType: ADDON_TYPE_EXTENSION,
+        category: 'causes',
+        query,
       },
     });
 
+    _fetchCategories({ store });
+    _loadCategories({ store });
+
     const root = render({ store });
 
     expect(root.find('.SearchContextCard-header')).toIncludeText(
-      '2 extensions found for "test"',
+      `2 extensions found for "${query}" in ${categoryName}`,
+    );
+  });
+
+  it('should render results with categoryName and no query for addonType ADDON_TYPE_EXTENSION when there is no query and when loading is false', () => {
+    const categoryName = 'Causes';
+    const { store } = dispatchSearchResults({
+      filters: {
+        addonType: ADDON_TYPE_EXTENSION,
+        category: 'causes',
+      },
+    });
+
+    _fetchCategories({ store });
+    _loadCategories({ store });
+
+    const root = render({ store });
+
+    expect(root.find('.SearchContextCard-header')).toIncludeText(
+      `2 extensions found in ${categoryName}`,
+    );
+  });
+
+  it('should render results without categoryName or query when neither are present for addonType ADDON_TYPE_EXTENSION', () => {
+    const { store } = dispatchSearchResults({
+      filters: {
+        addonType: ADDON_TYPE_EXTENSION,
+      },
+    });
+
+    const root = render({ store });
+
+    expect(root.find('.SearchContextCard-header')).toIncludeText(
+      '2 extensions found',
     );
   });
 
   it('should render singular form when only one result is found with addonType ADDON_TYPE_LANG', () => {
```

Comment:
As mentioned above, do we still need tests specifically for `ADDON_TYPE_LANG` when that is no longer being specifically dealt with in the code? Should we instead just have a set of tests for when the type is neither extension nor theme?